### PR TITLE
Skip mcg related test on openshift dedicated platform

### PR DESCRIPTION
**Pull request by suchita-g**
Signed-off-by: Suchita Gatfane <sgatfane@redhat.com>

### CODE REVIEW

**Comment by MeridianExplorer** [comment]
Why do those tests need to be skipped on Openshift Dedicated?

**Comment by suchita-g** [comment]
Marked this pr as `Verified`.  As  per expectation, all the tests in this PR verification job are [skipped](http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/sgatfane-osdtest1/sgatfane-osdtest1_20210706T092304/logs/test_report_1625589348.html)

**Comment by ebenahar** [comment]
> @suchita-g - Why do those tests need to be skipped on Openshift Dedicated?

MCG is disabled on OCS dedicated clusters. This might change in the future though

**Review by MeridianExplorer — CHANGES_REQUESTED** [review]
Assuming MCG is not available on Openshift Dedicated at all, all of the tests in the MCG directory need to be skipped.

I think the best way to do it would be by creating a `conftest.py` file in the `tests/manage/mcg/` directory, and to use `pytest_collection_modifyitems` to remove the test items.

An example can be seen [here](https://github.com/red-hat-storage/ocs-ci/blob/master/tests/manage/rgw/conftest.py)

**Comment by suchita-g** [comment]
> @suchita-g - Why do those tests need to be skipped on Openshift Dedicated?

Currently Openshift Dedicated platform i.e. ODF managed service does not support noobaa/mcg. So we are skipping all the test which involves mcg/noobaa

**Review by MeridianExplorer — DISMISSED** [review]
I do not think that the current skipping mechanisms are robust/futureproof enough, please see [this comment](https://github.com/red-hat-storage/ocs-ci/pull/4528#pullrequestreview-701843910)

**Review by fbalak — APPROVED** [review]

**Comment by suchita-g** [comment]
PR Verified  ref: https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/5541/

File: tests/manage/mcg/conftest.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    A pytest hook to filter out mcg tests
+    when running on openshift dedicated platform
+
+    Args:
+        items: list of collected tests
+
+    """
+    if (
+        config.ENV_DATA["platform"].lower() == OPENSHIFT_DEDICATED_PLATFORM
+        and float(config.ENV_DATA["ocs_version"]) < 4.8
```
Comment (ebenahar) [inline tests/manage/mcg/conftest.py:19]: will MCG be deployed with dedicated and OCS 4.8?

**Review by fbalak — APPROVED** [review]

File: tests/manage/mcg/conftest.py (hunk 1 above)
Comment (suchita-g) [reply tests/manage/mcg/conftest.py:19]: I checked with engineering for the exact version, the exact version is not yet decided. I am removing the condition from the code with TODO comment

**Review by fbalak — APPROVED** [review]

**Comment by suchita-g** [comment]
> PR validation on existing cluster
> 
> Cluster Name: sgatfane-workload
> Cluster Configuration: conf/ocsci/openshift_dedicated.yaml
> PR Test Suite: tier1
> PR Test Path: tests/manage/mcg tests/manage/storageclass/test_rbd_csi_default_sc.py::TestBasicPVCOperations::test_ocs_347
> Additional Test Params:
> OCP VERSION: 4.8
> OCS VERSION: 4.7
> 
> [Job UNSTABLE](https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-trigger-test-pr/1422//testReport) (some or all tests failed).

in this Job All MCG tests are removed from the list and the `test_ocs_347` test got executed. However, this test failed and hence Job. However, with respect to PR, this is expected behaviour.

**Review by ebenahar — APPROVED** [review]

**Pull request merged**